tests/unit/test_upload_handler.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 - 2022 Universität Tübingen, DKFZ and EMBL
```

Comment:
API level integration test is missing. Maybe we can discuss later. Just ping me when you have time.